### DISCUSSION TITLE
Common Defi Protocol for studying common  architecture

### DISCUSSION THREAD

**Discussion by sagoto80** [Cyfrin/security-and-auditing-full-course-s23#76] [Q&A] [answered]
There are  wide range of protocol in   competitive
audit. please suggest me protocols .

**Comment by johnsonstephan** [comment, marked as answer]
Hi @sagoto80, good question as there are dozens of options to choose from!

> Q: Could you please suggest some common DeFi protocols used for studying common architecture in competitive audits?

**TL;DR: Aave, Uniswap, and MakerDAO are three common protocols that are important for studying common architecture.**  

---

### Let's get into more detail!

The three recommended protocols are:

1. **Aave**: a leading lending protocol in the DeFi landscape that allows users to lend and borrow. It uses the native AAVE token for security and governance. Users can stake AAVE tokens to gain rewards
    - Studying the architecture of Aave can be done [here](https://docs.aave.com/developers/getting-started/readme) (and their latest audit report can be found [here](https://docs.aave.com/developers/deployed-contracts/security-and-audits)).
3. **Uniswap**: a decentralized exchange protocol that enables direct trading of ERC20 tokens. Liquidity providers supply assets to a liquidity pool and earn fees from trades. The native token UNI is earned through providing liquidity.
   - Studying Uniswap's architecture can be done [here](https://docs.uniswap.org/contracts/v3/overview) (and their latest audit report can be found [here](https://github.com/Uniswap/v3-core/blob/main/audits/tob/audit.pdf)).
5. **MakerDAO**: a decentralized credit platform that allows users to generate DAI, a stablecoin pegged to USD, by locking up collateral in the form of cryptocurrency. The protocol is governed by a community of MKR token holders.
   - You may study their architecture [here](https://docs.makerdao.com/getting-started/maker-protocol-101) (and their latest audit report can be found [here](https://security.makerdao.com/audit-reports)). 

These protocols have been extensively audited, providing valuable insights into their architectural design and potential vulnerabilities.

<details><summary>For further exploration, consider the following bonus resources.</summary>
<p>

**Aggregated Vulnerabilities**
- [Solodit](https://solodit.xyz/): a vulnerabilities aggregator used by top Web3 security auditors. You can explore reports, findings, and bugs from previous audits.
- [Blockchain Security Audit List](https://github.com/0xNazgul/Blockchain-Security-Audit-List): an aggregation of audit companies and their public audits.

**Competitive Audit / Bug Bounty Platforms**
- [CodeHawks](https://www.codehawks.com/contests): review their completed reports from various DeFi projects.
- [Code4Rena](https://code4rena.com/audits#completed-audits): review their completed audits of DeFi protocols.
- [Immunefi](https://immunefi.medium.com/): their blog is valuable resource for learning about vulnerabilities.

**News** (the latest and greatest incidents)
- [Rekt News](https://rekt.news/): a chronicle documenting compromised DeFi protocols. It's a useful resource to stay updated on the latest security incidents in DeFi.
- [BlockThreat](https://newsletter.blockthreat.io/archive): a weekly newsletter that shares security news, tools, events, vulnerabilities, and threats in the space.

**Hands-on Practice**
- [Blocksec CTFs](https://github.com/blockthreat/blocksec-ctfs): a curated list of blockchain security wargames, challenges, and capture the flag (CTF) competitions and solution write-ups. It's a great way to learn about security vulnerabilities and get hands-on experience with simulating actual scenarios.

</p>
</details> 


Does this answer your question?

**Reply by sagoto80** [reply]
thanks man .You are great

**Reply by johnsonstephan** [reply]
@sagoto80 Happy to help and glad it was useful!

**Comment by Avkubs** [comment]
Which of the three protocols is the future?
